flocker/node/script.py

```diff
@@ -396,6 +403,55 @@ class DeployerType(Names):
     block = NamedConstant()
 
 
+from flocker.node.agents.cinder import cinder_api
+from flocker.node.agents.test.blockdevicefactory import (
+    _openstack_auth_from_config
+)
+from keystoneclient.session import Session
+from cinderclient.client import Client as CinderClient
+from novaclient.client import Client as NovaClient
+
+
+def cinder_from_configuration(**config):
+    """
+    Build a ``CinderBlockDeviceAPI`` using configuration and
+    credentials in ``config``.
+    """
+    region = config.pop('region')
+    auth = _openstack_auth_from_config(**config)
+    session = Session(auth=auth)
+    cinder_client = CinderClient(
+        session=session, region_name=region, version=1
+    )
+    nova_client = NovaClient(
+        session=session, region_name=region, version=2
+    )
+
+    return cinder_api(
+        cinder_client=cinder_client,
+        nova_client=nova_client,
+        cluster_id=config['cluster_id']
+    )
```

Comment:
This function looks familiar.  Can you refactor this and `_openstack` from `blockdevicefactory.py` to remove the duplication?